pkg/controller/restore-operator/backup_handler.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"k8s.io/client-go/kubernetes"
+
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/backup"
+	"github.com/coreos/etcd-operator/pkg/backup/backend"
+	"github.com/coreos/etcd-operator/pkg/backup/backupapi"
+	backupS3 "github.com/coreos/etcd-operator/pkg/backup/s3"
+	"github.com/coreos/etcd-operator/pkg/cluster/backupstorage"
+	"github.com/sirupsen/logrus"
+)
+
+// backupServerCloser wraps around backup.BackupServer with a close method
+// that cleans up resources associated with backup.BackupServer.
+type backupServerCloser struct {
```

Comment:
just backupServer. closer has no meaning here. no one will use it externally as an interface.